src/Processor/Consumers/CustomsDeclarationsConsumer.cs

```
@@ -35,14 +35,26 @@ CancellationToken cancellationToken
 
         var existingValidationErrors = await api.GetProcessingError(mrn, cancellationToken);
 
+        validationResult.Errors.ForEach(error =>
+            logger.LogInformation(
+                "Mrn {Mrn} Version {Version} Type {MessageType} failed validation with {ErrorCode}: {ErrorMessage}",
+                mrn,
+                customsDeclarationsMessage.Header.EntryVersionNumber,
+                messageType,
+                error.ErrorCode,
+                error.ErrorMessage
+            )
+        );
+
         var processingErrorNotification = new DataApiErrors.ErrorNotification
         {
             ExternalCorrelationId = customsDeclarationsMessage.ServiceHeader.CorrelationId,
             ExternalVersion = customsDeclarationsMessage.Header.EntryVersionNumber,
             Errors = validationResult
-                .Errors.Select(error => new DataApiErrors.ErrorItem
+                .Errors.Where(error => error.CustomState != null)
+                .Select(error => new DataApiErrors.ErrorItem
```

Do we lose an error here if `CustomState` is null and `ErrorMessage` is not?

---

Discussed - this is where errors are discarded and the resultant set that is saved against the data API